toolchain/check/testdata/interop/cpp/function/class.carbon

```diff
@@ -211,6 +211,58 @@ fn F() {
   //@dump-sem-ir-end
 }
 
+// ============================================================================
+// Defined class in class
+// ============================================================================
+
+// --- definition_in_outer_definition.h
+
+class O {
+ public:
+  class C {};
+};
+
+auto foo(O::C) -> void;
+
+// --- fail_todo_import_definition_in_outer_definition.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "definition_in_outer_definition.h";
+
+fn F() {
+  //@dump-sem-ir-begin
+  // CHECK:STDERR: fail_todo_import_definition_in_outer_definition.carbon:[[@LINE+7]]:3: error: semantics TODO: `Unsupported mapping of a C++ record to a type within a declaration context that is not the translation unit or a namespace` [SemanticsTodo]
+  // CHECK:STDERR:   Cpp.foo({});
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR: fail_todo_import_definition_in_outer_definition.carbon:[[@LINE+4]]:3: note: in `Cpp` name lookup for `foo` [InCppNameLookup]
+  // CHECK:STDERR:   Cpp.foo({});
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  Cpp.foo({});
+  //@dump-sem-ir-end
+}
+
+// --- fail_todo_import_definition_in_outer_definition_and_use_outer.carbon
+
+library "[[@TEST_NAME]]";
+
+import Cpp library "definition_in_outer_definition.h";
+
+fn F() {
+  // CHECK:STDERR: fail_todo_import_definition_in_outer_definition_and_use_outer.carbon:[[@LINE+7]]:3: error: semantics TODO: `Unsupported mapping of a C++ record to a type within a declaration context that is not the translation unit or a namespace` [SemanticsTodo]
+  // CHECK:STDERR:   Cpp.foo({});
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR: fail_todo_import_definition_in_outer_definition_and_use_outer.carbon:[[@LINE+4]]:3: note: in `Cpp` name lookup for `foo` [InCppNameLookup]
+  // CHECK:STDERR:   Cpp.foo({});
+  // CHECK:STDERR:   ^~~~~~~
+  // CHECK:STDERR:
+  Cpp.foo({});
+  //@dump-sem-ir-begin
+  var x: Cpp.O;
```

Comment:
Since the second check is no longer calling, I think it'd make sense to re-combine the two tests now. It just hadn't been clear to me what you intended to test before: if you were testing the type was usable post-`Cpp.foo`, or if you were testing that a type containing a type could be used at all.

Reply:
Done.